src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -271,6 +290,11 @@ private PssaCmdletAnalysisEngine InstantiateAnalysisEngine()
             if (TryFindSettingsFile(out string settingsFilePath))
             {
                 _logger.LogInformation($"Configuring PSScriptAnalyzer with rules at '{settingsFilePath}'");
+                _pssaSettingsFileWatcher = new FileSystemWatcher(settingsFilePath)
+                {
+                    EnableRaisingEvents = true,
+                };
+                _pssaSettingsFileWatcher.Changed += OnSettingsFileUpdated;
```

Comment:
I don't see any code that unsubscribes before it gets disposed.
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-subscribe-to-and-unsubscribe-from-events#unsubscribing